core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -333,6 +346,181 @@ public long nextFetchOffset() {
         }
     }
 
+    private void updateOffsetsOnLsoMovement() {
+        long logStartOffset = offsetForEarliestTimestamp();
+        Throwable throwable = null;
+        if (logStartOffset > startOffset) {
```

Comment:
Isn't startOffset a shared variable and atleast a read lock should occur for that?